drivers/spi/spi_mcux_flexio.c

```diff
@@ -100,6 +100,16 @@ static int spi_mcux_flexio_isr(void *user_data)
 	const struct spi_mcux_flexio_config *config = dev->config;
 	struct spi_mcux_flexio_data *data = dev->data;
 
+#if defined(CONFIG_SOC_SERIES_KE1XZ)
```

Comment:
why does this not apply to other devices?

Reply:
I don't know if this works on other platforms, I haven't had a chance to test other platforms.

Reply:
I don't think is correct to wait on the ISR for `kFLEXIO_SPI_TxBufferEmptyFlag` to be cleared. The interrupt is raised on async transfers and `FLEXIO_SPI_MasterTransferHandleIRQ` is checking for this flag already.
What is the reason you are waiting for this flag to be cleared?

Reply:
It doesn't work on my site, the program will get an exception without waiting.
In `FLEXIO_SPI_MasterTransferHandleIRQ`, the status of flexio is indeed detected. However, it is possible that the state has not been stabilized and some data has not been successfully transmitted. So I added the loop to wait for the state to change before calling this function.

![image](https://github.com/user-attachments/assets/eba7f95c-80eb-42d2-af85-2e170f3ec409)

Reply:
And `FLEXIO_SPI_TransferSendTransaction` function calls `FLEXIO_SPI_WriteData` to write data. The TxEmptyFlag needs to be asserted before calling this API. See the description in fsl_flexio.h driver:
![image](https://github.com/user-attachments/assets/a5ef0791-3f86-4f39-a0ff-1835269f9b4d)